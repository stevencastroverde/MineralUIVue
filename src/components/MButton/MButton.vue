<template>
    <button 
    class="m-button" 
    :class="computedClass"
    :disabled="disabled"
    >
        <slot></slot>
    </button>
</template>

<script>
export default {
    name: 'MButton',
    props: {
        disabled: {
            type: Boolean,
            default: false,
        },
        flat: {
            type: Boolean,
            default: false,
        },
        success: {
            type: Boolean,
            default: false,
        }
    },
    computed: {
        computedClass() {
            return {
                'm-button--flat': this.flat,
                'm-button--success': this.success
            }
        }
    }
}
</script>

<style lang="scss">
    .m-button {
        line-height: 1.25;
        min-width: 2.5em;
        height: 2.5em;
        border-radius: .15em;
        border-color: #c8d1e0;
        border-width: 1px;
        font-weight: 600;
        padding: 0 .5em;
        font-size: 1rem;
        color: #3272d9;
        transition: all .1s ease-in;
        &--success {
            color: #2a854e;
        }
        &--flat {
            border-color: transparent;
        }
        &:hover {
            background-color: rgba(#222, .1);
        }
        &:disabled {
            color: #afbacc;
            background-color: #ebeff5;
        }
    }
</style>