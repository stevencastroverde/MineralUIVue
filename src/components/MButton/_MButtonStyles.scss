.m-button {
    $self: &;
    line-height: 1.25;
    min-width: 2.5em;
    height: 2.5em;
    border-radius: .15em;
    border-color: #c8d1e0;
    border-width: 1px;
    font-weight: 600;
    outline: 0;
    padding: 0 .5em;
    font-size: 1rem;
    color: #3272d9;
    transition: all .1s ease-in;
    &:hover {
        background-color: #f5f7fa;
        border-color: #3272d9;
    }
    &:disabled {
        color: #afbacc;
        background-color: #ebeff5;
        border-color: transparent;
    }
    &--success {
        color: #2a854e;
        &:hover {
            border-color: #2a854e;
        }
        &:focus {
            box-shadow: rgb(255, 255, 255) 0px 0px 0px 1px, #2a854e 0px 0px 0px 2px;
        }
        &:active {
            background-color: #ebeff5;
            border-color: darken(#2a854e, 2);
        }
    }
    &--danger {
        color: #de1b1b;
        &:hover {
            border-color: #de1b1b;
        }
        &:focus {
            box-shadow: rgb(255, 255, 255) 0px 0px 0px 1px, #de1b1b 0px 0px 0px 2px;
        }
    }
    &--warning {
        color: #ad5f00;
        &:hover {
            border-color: #ad5f00;
        }
    }
    &--minimal {
        border-color: transparent;
        &:hover {
            border-color: transparent;
        }
        &:active {
            background-color: #ebeff5;
        }
        &:disabled {
            background-color: transparent;
        }
    }
    // sizing 
    &--small {
        width: 1em;
        height: 1em;
    }
    &--medium {
        width: 2.5em;
        height: 2.5em;
    }
    &--large {
        width: 4em;
        height: 4em;
    }
    &--jumbo {
        width: 6em;
        height: 6em;
    }
    &--full-width {
        width: 100%
    }
    &--circular {
        display: inline-block;
        border-radius: 50%;
        width: 2.5em;
        line-height: 1;
        >* {
            padding: 0 !important;
            margin: 0 !important;
        }
    }
    //icons
    [role="presentation"]:first-child {
        margin-right: 0.5em;
    }
    [role="presentation"]:last-child {
        margin-left: 0.5em;
    }
}