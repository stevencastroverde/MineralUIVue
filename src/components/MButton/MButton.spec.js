import { mount, shallowMount } from '@vue/test-utils';
import MButton from './MButton.vue';
import IconAccountBox from '../Icons/IconAccountBox.vue';

describe('MButton', () => {
    it('renders correctly', () => {
        const wrapper = mount(MButton, {
            slots: {
                default: 'Test Button',
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    it('creates icon component', () => {
        const wrapper = mount(MButton, {
            propsData: {
                appendIcon: 'IconAccountBox',
            },
        });
        debugger;
        expect(wrapper.contains(IconAccountBox)).toBe(true);
    })






})