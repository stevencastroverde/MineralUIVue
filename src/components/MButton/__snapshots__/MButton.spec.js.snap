// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MButton renders correctly 1`] = `
<button
  class="m-button"
  style="font-size: 1em; padding: 0px .5em;"
>
  <!---->
   Test Button 
  <!---->
</button>
`;
